konrad/radiation/rrtmg.py

```diff
@@ -262,6 +268,44 @@ def radiative_fluxes(self, atmosphere, surface, cloud):
 
         return lw_fluxes, sw_fluxes
 
+    def calc_cloudy_nomcica_radiation(self, atmosphere, surface, cloud):
+
+        cloud_fraction = deepcopy(cloud.cloud_area_fraction_in_atmosphere_layer)
+
+        if self._state_sw is None:  # first time only
+            cf_cloudy = cloud_fraction[cloud_fraction != 0]
+            if cf_cloudy.shape[0] != 0:
+                if not np.all(cf_cloudy == cf_cloudy[0]):
+                    logging.warning(
+                        'The konrad implementation of nomcica with partial '
+                        'cloud fraction is only suitable for a single '
+                        'rectangular cloud. Consider using mcica instead.')
+
+        cf_max = np.max(cloud_fraction)
+
+        # Calculate overcast and clear sky radiative fluxes.
+        # Make all the cloudy layers overcast, so that the nomcica version of
+        # RRTMG can be used in the shortwave - all wavelengths see cloud.
+        # Use the same approach for the longwave for consistency.
+        cloud.cloud_area_fraction_in_atmosphere_layer[cloud_fraction != 0] = 1
+        lw_overcast, sw_overcast = self.radiative_fluxes(
+            atmosphere, surface, cloud
+        )
+        lw_fluxes, sw_fluxes = lw_overcast[1], sw_overcast[1]
+
+        # Combine overcast and clear sky fluxes and heating rates to get the
+        # all sky fluxes and heating rates.
+        for fluxes in lw_fluxes, sw_fluxes:
+            for key in fluxes.keys():
+                if 'clear_sky' not in key:
+                    clear_part = fluxes[key + '_assuming_clear_sky'][:]
+                    fluxes[key][:] *= cf_max  # weighted by cloud area fraction
+                    fluxes[key][:] += (1 - cf_max) * clear_part
```

Comment:
Does this give you the net flux? If yes, does the loop overwrite both the net and the cloudy fluxes with the net fluxes?

Reply:
Yes, this gives the net flux. The fluxes (and heating rates) stored are just for the clear sky and the net. In this method, we first calculate overcast fluxes, which are stored in the net flux variables and then over-right them with the actual net fluxes.